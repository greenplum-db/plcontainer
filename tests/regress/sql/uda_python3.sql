-- Test UDA with prefunc
CREATE FUNCTION sfunc_sum_cols(state numeric, col_a numeric, col_b numeric) 
RETURNS numeric AS 
$$
# container: plc_python_shared
   return state + col_a + col_b
$$ language plcontainer;


CREATE FUNCTION prefunc_sum_cols(state_a numeric, state_b numeric) 
RETURNS numeric AS 
$$
# container: plc_python_shared
   return state_a + state_b
$$ language plcontainer;

CREATE FUNCTION final_func_sum_cols(state_a numeric)
RETURNS numeric AS
$$
# container: plc_python_shared
   return state_a*10
$$ language plcontainer;

CREATE AGGREGATE sum_cols(numeric, numeric) (
	   SFUNC = sfunc_sum_cols,
	   PREFUNC = prefunc_sum_cols,
	   FINALFUNC = final_func_sum_cols,
	   STYPE = numeric,
	   INITCOND = 0 
);

create table t (i int, j int);
insert into t values(2,1),(4,8),(9,10),(10,24),(44,11);
select sum_cols(i,j) FROM t;
drop table t;
