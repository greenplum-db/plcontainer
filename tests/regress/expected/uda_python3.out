-- Test UDA with prefunc
CREATE FUNCTION sfunc_sum_cols(state numeric, col_a numeric, col_b numeric) 
RETURNS numeric AS 
$$
# container: plc_python_shared
   return state + col_a + col_b
$$ language plcontainer;
CREATE FUNCTION prefunc_sum_cols(state_a numeric, state_b numeric) 
RETURNS numeric AS 
$$
# container: plc_python_shared
   return state_a + state_b
$$ language plcontainer;
CREATE FUNCTION final_func_sum_cols(state_a numeric)
RETURNS numeric AS
$$
# container: plc_python_shared
   return state_a*10
$$ language plcontainer;
CREATE AGGREGATE sum_cols(numeric, numeric) (
	   SFUNC = sfunc_sum_cols,
	   PREFUNC = prefunc_sum_cols,
	   FINALFUNC = final_func_sum_cols,
	   STYPE = numeric,
	   INITCOND = 0 
);
create table t (i int, j int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into t values(2,1),(4,8),(9,10),(10,24),(44,11);
select sum_cols(i,j) FROM t;
 sum_cols 
----------
     1230
(1 row)

drop table t;
